From 1f5f51b80b1f0c9bf52566e8174aa617921a4e82 Mon Sep 17 00:00:00 2001
From: Felipe Balbi <balbi@ti.com>
Date: Tue, 15 Sep 2015 09:49:14 -0500
Subject: [PATCH 32/78] usb: dwc3: gadget: clear DWC3_PENDING_REQUEST when
 request is queued

Instead of clearing DWC3_PENDING_REQUEST when
we start transfer, let's do it when the request
is actually queued, that way we know for sure
that we're clearing in the right time.

Signed-off-by: Felipe Balbi <balbi@ti.com>
(cherry picked from commit 89185916d2295f7c4bc3a05adae25380b3e08e58)
Signed-off-by: Janusz Dziedzic <januszx.dziedzic@intel.com>

Change-Id: I0c5130ef5e6e44091c4ee67d0553653229c2f2a6
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-19337
Reviewed-on: https://android.intel.com:443/511157
---
 drivers/usb/dwc3/gadget.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index 7a8932d..8a81179 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -949,7 +949,6 @@ static int __dwc3_gadget_kick_transfer(struct dwc3_ep *dep, u16 cmd_param,
 		dwc3_trace(trace_dwc3_gadget, "%s: endpoint busy", dep->name);
 		return -EBUSY;
 	}
-	dep->flags &= ~DWC3_EP_PENDING_REQUEST;
 
 	/*
 	 * If we are getting here after a short-out-packet we don't enqueue any
@@ -1168,6 +1167,10 @@ static int __dwc3_gadget_ep_queue(struct dwc3_ep *dep, struct dwc3_request *req)
 		if (ret && ret != -EBUSY)
 			dwc3_trace(trace_dwc3_gadget, "%s: failed to kick transfers",
 					dep->name);
+
+		if (!ret)
+			dep->flags &= ~DWC3_EP_PENDING_REQUEST;
+
 		return ret;
 	}
 
-- 
1.9.1

